tests/foreman/cli/test_classparameters.py

```diff
@@ -154,9 +151,16 @@ def test_positive_list_with_non_admin_user(self, session_puppet_enabled_sat, mod
                 ]
             },
         }
-        user = make_user({'admin': '0', 'password': password})
-        role = make_role()
-        add_role_permissions(role['id'], required_user_permissions)
+        user = session_puppet_enabled_sat.cli_factory.make_user(
+            {'admin': '0', 'password': password}
+        )
+        role = session_puppet_enabled_sat.cli_factory.make_role()
+        # session_puppet_enabled_sat.cli_factory.add_role_permissions(
+        #     role['id'], required_user_permissions
+        # )
+        with session_puppet_enabled_sat:
+            add_role_permissions(role['id'], required_user_permissions)
```

Comment:
Here I had to use context manager for `add_role_permissions`, because the way with mere `session_puppet_enabled_sat.cli_factory.add_role_permissions` fails in https://github.com/SatelliteQE/robottelo/blob/8c27c7d5ed955f7034ab8a8ca82afbeece417973/robottelo/host_helpers/cli_factory.py#L1177
with
```
>           self.make_filter(options=options)
E           TypeError: 'NoneType' object is not callable
```
as `filter` has no dict items.

Another ugly fix is to add `make_filter` method to the `CLIFactory` class
```
    def make_filter(self, options=None):
        return create_object(self._satellite.cli.Filter, options)
```
but this is not the path we would like to follow.

Any ideas what is the correct approach here @JacobCallahan ?